metadata-ingestion/src/datahub/ingestion/run/pipeline.py

```diff
@@ -101,6 +106,16 @@ class PipelineInitError(Exception):
     pass
 
 
+@contextlib.contextmanager
+def _add_init_error_context(step: str) -> Iterator[None]:
+    """Enriches any exceptions raised with information about the step that failed."""
+
+    try:
+        yield
+    except Exception as e:
+        raise PipelineInitError(f"Failed to {step}: {e}") from e
```

Comment:
Failed while executing step {step}?

Reply:
This keeps the error messages the same as before e.g. "Failed to set up framework context" - i didn't want to make the errors too verbose